app/validators/terms_are_importable.rb

```diff
@@ -0,0 +1,35 @@
+# Validator for an array of terms to ensure none of them have any "core"
+# validation errors
+class TermsAreImportable < ActiveModel::Validator
+  def validate(record)
+    record.terms.each do |term|
+      term = decorate(term)
+      unless term.valid?
+        record.errors.add(:base, term_errors(term))
```

Comment:
I keep wanting to, but I didn't really build it for collections - and I'm not sure it's worth the pain of making a composite for error objects.  Additionally, I would need to allow for a prefix, rather than a raw error propagation situation ("opaquenamespace.org/vocab/id is not importable: ...").  And even then I'd need to add something to turn multiple errors into a single comma-delimited list.

I think the solution would be a customizable error composite which TermsAreImportable would then use in a call to the fairly simple ErrorPropagator.  But I don't know if that's a worthwhile change right now.